client/allocrunner/taskrunner/task_runner.go

```diff
@@ -360,6 +370,10 @@ func (tr *TaskRunner) initLabels() {
 // Run the TaskRunner. Starts the user's task or reattaches to a restored task.
 // Run closes WaitCh when it exits. Should be started in a goroutine.
 func (tr *TaskRunner) Run() {
+	// Mark that the run routine has been launched so that other functions can
+	// decide to use the wait channel or not.
+	tr.setRunLaunched()
```

Comment:
This is racy since TR.Run is called in a goroutine. To eliminate the race we need to make TR.Run synchronous just so that runLaunched is set before the caller continues:

```go
func (tr *TaskRunner) Run() {
	tr.setRunLaunched()

	go tr.run() // all of the below code
}
```

Reply:
Talked offline and we agree this isn't necessary due to kill ctx being canceled and the run loop no-oping in this case